### ISSUE TITLE
ERROR: for bootstrap

### ISSUE THREAD

**Issue by derekkinsman** [konradkonrad/docker-pyeth-cluster#3]
I get this error when trying to run the `with-netstats` setup. This is running on an AWS EC2 Ubuntu 14.04 instance.

```
ERROR: Service 'bootstrap' failed to build: The command '/bin/sh -c apt-get update &&    apt-get install -y curl git-core &&    curl -sL https://deb.nodesource.com/setup | bash - &&    apt-get update &&    apt-get install -y nodejs' returned a non-zero code: 1
```

Here's the whole log if you'd like to see it:

```
Σ:~/ethereum-cluster/docker-pyeth-cluster/with-netstats$ sudo docker-compose scale bootstrap=1 miner=2 eth=10 statsmon=1
WARNING: The "bootstrap" service specifies a port on the host. If multiple containers for this service are created on a single host, the port will clash.
Creating and starting bootstrap ...
Building bootstrap
Step 1 : FROM ethereum/client-python
latest: Pulling from ethereum/client-python
e5ad7970bc69: Pull complete
a3ed95caeb02: Pull complete
0a7eeedff1c5: Pull complete
bbe5878d8681: Pull complete
1df3bb332f6a: Pull complete
9b26172f72dc: Pull complete
66827c4496e3: Pull complete
d53e9c58fa7d: Pull complete
9b17be6e2a41: Pull complete
667f7559b8f7: Pull complete
3cb99d002509: Pull complete
8cb930eca1f0: Pull complete
eb93cf4ba8bc: Pull complete
bbdb17f4e286: Pull complete
bd5661131733: Pull complete
b20f9a854d53: Pull complete
d97aa8e8ee15: Pull complete
b77ac10b88bf: Pull complete
197993d2de75: Pull complete
d35b63a73b82: Pull complete
05e4cd38daa8: Pull complete
1b08ec3eb780: Pull complete
Digest: sha256:aa1da3dc2f88140df4ef5cfdd95f4d75670308f2e1c554d7c50da016b9171f48
Status: Downloaded newer image for ethereum/client-python:latest
 ---> 1723d972bbea
Step 2 : RUN apt-get update &&    apt-get install -y curl git-core &&    curl -sL https://deb.nodesource.com/setup | bash - &&    apt-get update &&    apt-get install -y nodejs
 ---> Running in 4768a10aa4a1
Get:1 http://security.debian.org jessie/updates InRelease [63.1 kB]
Get:2 http://security.debian.org jessie/updates/main amd64 Packages [300 kB]
Ign http://httpredir.debian.org jessie InRelease
Get:3 http://httpredir.debian.org jessie-updates InRelease [142 kB]
Get:4 http://httpredir.debian.org jessie Release.gpg [2373 B]
Get:5 http://httpredir.debian.org jessie-updates/main amd64 Packages [9283 B]
Get:6 http://httpredir.debian.org jessie Release [148 kB]
Get:7 http://httpredir.debian.org jessie/main amd64 Packages [9034 kB]
Fetched 9700 kB in 14s (669 kB/s)
Reading package lists...
W: Size of file /var/lib/apt/lists/httpredir.debian.org_debian_dists_jessie_main_binary-amd64_Packages.gz is not what the server reported 9033871 9035431
Reading package lists...
Building dependency tree...
Reading state information...
The following packages were automatically installed and are no longer required:
  file gir1.2-freedesktop gir1.2-gdkpixbuf-2.0 gir1.2-glib-2.0 gir1.2-rsvg-2.0
  javascript-common libcairo-gobject2 libcairo-script-interpreter2 libcdt5
  libcgraph6 libcroco3 libdjvulibre-dev libelfg0 libexif-dev libexif12
  libexpat1-dev libfontconfig1-dev libfreetype6-dev libgd3
  libgirepository-1.0-1 libglib2.0-bin libgraphviz-dev libgvc6 libgvpr2
  libice-dev libice6 libilmbase-dev libjasper-dev libjbig-dev libjs-excanvas
  libjs-jquery liblcms2-dev liblqr-1-0-dev libltdl-dev liblzo2-2 libmagic1
  libmagickcore-6-arch-config libmagickcore-6-headers libmagickwand-6-headers
  libopenexr-dev libpathplan4 libpcre3-dev libpcrecpp0 libpixman-1-dev
  libpng12-dev libpthread-stubs0-dev librsvg2-2 librsvg2-common libsm-dev
  libsm6 libtiff5-dev libtiffxx5 libtool libvpx1 libwmf-dev libx11-dev
  libx11-doc libxau-dev libxcb-render0-dev libxcb-shm0-dev libxcb1-dev
  libxdmcp-dev libxdot4 libxext-dev libxpm4 libxrender-dev libxt-dev libxt6
  mime-support pkg-config x11-common x11proto-core-dev x11proto-input-dev
  x11proto-kb-dev x11proto-render-dev x11proto-xext-dev xorg-sgml-doctools
  xtrans-dev
Use 'apt-get autoremove' to remove them.
The following extra packages will be installed:
  libcurl3 libcurl4-openssl-dev
Suggested packages:
  libcurl4-doc libcurl3-dbg libidn11-dev libkrb5-dev libldap2-dev librtmp-dev
  libssh2-1-dev
The following packages will be upgraded:
  curl git-core libcurl3 libcurl4-openssl-dev
4 upgraded, 0 newly installed, 0 to remove and 114 not upgraded.
Need to get 796 kB of archives.
After this operation, 21.5 kB of additional disk space will be used.
Get:1 http://httpredir.debian.org/debian/ jessie/main libcurl4-openssl-dev amd64 7.38.0-4+deb8u3 [336 kB]
Get:2 http://httpredir.debian.org/debian/ jessie/main curl amd64 7.38.0-4+deb8u3 [200 kB]
Get:3 http://httpredir.debian.org/debian/ jessie/main libcurl3 amd64 7.38.0-4+deb8u3 [259 kB]
Get:4 http://httpredir.debian.org/debian/ jessie/main git-core all 1:2.1.4-2.1+deb8u2 [1496 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 796 kB in 0s (1364 kB/s)
(Reading database ... 25744 files and directories currently installed.)
Preparing to unpack .../libcurl4-openssl-dev_7.38.0-4+deb8u3_amd64.deb ...
Unpacking libcurl4-openssl-dev:amd64 (7.38.0-4+deb8u3) over (7.38.0-4+deb8u2) ...
Preparing to unpack .../curl_7.38.0-4+deb8u3_amd64.deb ...
Unpacking curl (7.38.0-4+deb8u3) over (7.38.0-4+deb8u2) ...
Preparing to unpack .../libcurl3_7.38.0-4+deb8u3_amd64.deb ...
Unpacking libcurl3:amd64 (7.38.0-4+deb8u3) over (7.38.0-4+deb8u2) ...
Preparing to unpack .../git-core_1%3a2.1.4-2.1+deb8u2_all.deb ...
Unpacking git-core (1:2.1.4-2.1+deb8u2) over (1:2.1.4-2.1) ...
Setting up libcurl3:amd64 (7.38.0-4+deb8u3) ...
Setting up libcurl4-openssl-dev:amd64 (7.38.0-4+deb8u3) ...
Setting up curl (7.38.0-4+deb8u3) ...
Setting up git-core (1:2.1.4-2.1+deb8u2) ...
Processing triggers for libc-bin (2.19-18) ...

## Installing the NodeSource Node.js 0.10 repo...


## Populating apt-get cache...

+ apt-get update
Hit http://security.debian.org jessie/updates InRelease
Get:1 http://security.debian.org jessie/updates/main amd64 Packages [300 kB]
Hit http://httpredir.debian.org jessie-updates InRelease
Ign http://httpredir.debian.org jessie InRelease
Err http://httpredir.debian.org jessie Release.gpg
  Error reading from server. Remote end closed connection [IP: 5.153.231.35 80]
Get:2 http://httpredir.debian.org jessie-updates/main amd64 Packages [9283 B]
Hit http://httpredir.debian.org jessie Release
Get:3 http://httpredir.debian.org jessie/main amd64 Packages [9034 kB]
Fetched 9343 kB in 3s (2854 kB/s)
W: Failed to fetch http://httpredir.debian.org/debian/dists/jessie/Release.gpg  Error reading from server. Remote end closed connection [IP: 5.153.231.35 80]

E: Some index files failed to download. They have been ignored, or old ones used instead.
Error executing command, exiting

ERROR: for bootstrap
ERROR: Service 'bootstrap' failed to build: The command '/bin/sh -c apt-get update &&    apt-get install -y curl git-core &&    curl -sL https://deb.nodesource.com/setup | bash - &&    apt-get update &&    apt-get install -y nodejs' returned a non-zero code: 1
```

**Comment by konradkonrad** [comment]
Hey @derekkinsman!
Can you make sure that this wasn't a temporary network issue?
`Failed to fetch http://httpredir.debian.org/debian/dists/jessie/Release.gpg  Error reading from server. Remote end closed connection [IP: 5.153.231.35 80]` makes it look like one.

**Comment by derekkinsman** [comment]
Hey @konradkonrad, I've been messing around with this a bit more and it looks like there's issues on AWS with binding IPs to 127.0.0.1 and 0.0.0.0.

When I switch everything to `0.0.0.0` netstats starts running but the network will never see the minimum required number of nodes to start verifying mining/transactions. If I run it on `127.0.0.1` netstats doesn't work, but the rest tof the network spins up and does what it is supposed to do.

I think it might be related to the recent changes that have made `--fake` not work. But I also have no idea and am fairly new to this. (I also had to change the source of the Nodesource install as NodeJS and IOJS have merged back together and they renamed all the distributions)

**Comment by konradkonrad** [comment]
Hmm. 
So this particular issue (ERROR: for bootstrap) seems to be resolved, right? Please close this issue, if your original problem is no longer an issue!

Can you share details for the nodesource install issue, or even better, create a PR for what is necessary there? I hadn't had an issue there today, most likely I got some things still cached here.

I assume, you're running all nodes in the same machine/instance, correct? Can you open a new issue for the connectivity issue on AWS, and give some more details about what works and what doesn't?

Btw, I am not sure, if `--fake` is really related to this. There were some issues with the `genesis_hash` that I resolved with #7. But this should not have resulted in any other networking issues. Please also test with the newest master. Thanks!

**Comment by derekkinsman** [comment]
Will do. Thanks! (I'll test the latest version of master before opening a new issue)
